lib/spack/spack/abi.py

```diff
@@ -52,12 +53,16 @@ def _gcc_get_libstdcxx_version(self, version):
         rungcc = None
         libname = None
         output = None
+        if sys.platform == "darwin":
+            suffix = "dylib"
+        else:
+            suffix = "so"
         if compiler.cxx:
             rungcc = Executable(compiler.cxx)
-            libname = "libstdc++.so"
+            libname = "libstdc++." + suffix
```

Comment:
inside `package.py` one can use `dso_suffix`. Not sure from where is it imported, but may work here as well. You won't need those ifs then.